components/uid.rst

```diff
@@ -0,0 +1,155 @@
+.. index::
+   single: UID
+   single: Components; UID
+
+The UID Component
+=================
+
+    The UID component provides utilities to work with `unique identifiers`_ (UIDs)
+    such as UUIDs and ULIDs.
+
+.. versionadded:: 5.1
+
+    The UID component was introduced in Symfony 5.1 as an
+    :doc:`experimental feature </contributing/code/experimental>`.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/uid
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+This component provides different classes to generate each type of UID. The
+following sections explain each of them.
+
+UUIDs
+-----
+
+`UUIDs`_ (*universally unique identifiers*) are one of the most popular UIDs in
+the software industry. UUIDs are 128-bit numbers usually represented as five
+groups of hexadecimal characters: ``xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx``
+(the ``M`` digit is the UUID version and the ``N`` digit is the UUID variant).
+
+Generating UUIDs
+~~~~~~~~~~~~~~~~
+
+Use any of the following named constructors depending on your needs::
+
+    use Symfony\Component\Uid\Uuid;
+
+    // UUID type 1 generates the UUID using the MAC address of your device and a timestamp.
+    // Both are obtained automatically, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v1();
+
+    // UUID type 4 generates a random UUID, so you don't have to pass any constructor argument.
+    $uuid = Uuid::v4();
+
+    // UUID type 3 and 5 generate a UUID hashing the given namespace and name. Type 3 uses
+    // MD5 hashes and Type 5 uses SHA-1. The namespace is another UUID (e.g. a Type 4 UUID)
+    // and the name is an arbitrary string (e.g. a product name; if it's unique).
+    $namespace = Uuid::v4();
+    $name = $product->getUniqueName();
+    $uuid = Uuid::v3($namespace, $name);
+    $uuid = Uuid::v5($namespace, $name);
+
+If your UUID is generated by another system, use the regular constructor of the
+``Uuid`` class to create an object and make use of the utilities available for
+Symfony UUIDs::
+
+    // this value is generated somewhere else
+    $uuidValue = 'd9e7a184-5d5b-11ea-a62a-3499710062d0';
+    $uuid = new Uuid($uuidValue);
+
+If your UUIDs are generated in binary format, use the ``fromBinary()`` method
+to create the objects for them::
+
+    $uuid = Uuid::fromBinary($uuidBinaryContents);
+
+    // the opposite method is also available
+    $uuidAsBinary = $uuid->toBinary();
+
+Working with UUIDs
+~~~~~~~~~~~~~~~~~~
+
+UUID objects created with the ``Uuid`` class can use the following methods
+(which are equivalent to the ``uuid_*()`` method of the PHP extension)::
+
+    use Symfony\Component\Uid\Uuid;
+
+    $uuid4 = Uuid::v4();
+
+    $uuid4->isNull(); // false
+
+    // the Uuid class defines public constants for all versions (Uuid::TYPE_1, Uuid::TYPE_3, etc.)
+    $uuid4->getType(); // int(4)
```

Comment:
this one is also removed. I guess both are now replaced by checking instance: `$uuid4 instanceof NilUuid` and `$uuid instanceof UuidV4`